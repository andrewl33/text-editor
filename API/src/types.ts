export interface IToken {
  user: string;
  collections: string[];
  files: string[];
}