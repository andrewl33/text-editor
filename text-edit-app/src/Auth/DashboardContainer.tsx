import { push, RouterAction } from "connected-react-router";
import * as React from "react";
import { connect } from "react-redux";
import { ThunkDispatch } from "redux-thunk";
import { HeaderComponent } from "../generic/TopBar/HeaderComponent";
import { DashboardContainerProps, HeaderProps, StoreState } from "../types";
import {
  AuthAction,
  closeAlert,
  closePrompt,
  logIn,
  logInPrompt,
  logOut
} from "./AuthAction";
import { DashboardComponent } from "./DashboardComponent";

export class DashboardContainer extends React.Component<
  DashboardContainerProps & HeaderProps
> {
  public render() {
    const {
      accountName,
      authPrompt,
      loggedIn,
      openAlert,
      alertMessage,
      onAlert,
      dashboard,
      onLogInPrompt,
      onLogOut,
      onFileClick,
      onCollectionClick,
      onAuthAccount,
      onClosePrompt,
      onHomeClick
    } = this.props;
    const header = (
      <HeaderComponent
        accountName={accountName}
        onHomeClick={onHomeClick}
        loggedIn={loggedIn}
        isShareable={false}
        openAlert={openAlert}
        alertMessage={alertMessage}
        pageName={"Dashboard"}
        onAlert={onAlert}
        onPrompt={authPrompt}
        onLogInPrompt={onLogInPrompt}
        onLogOut={onLogOut}
        prompt={"Login"}
        getAccountCredentials={onAuthAccount}
        onClosePrompt={onClosePrompt}
      />
    );

    if (!loggedIn || !dashboard) {
      return (
        <div>
          {header}
          <h1 style={{ color: "rgba(255,255,255,.9)" }}>Not logged in!</h1>
        </div>
      );
    } else {
      return (
        <div>
          {header}
          <DashboardComponent
            dashboard={dashboard}
            onFileClick={onFileClick}
            onCollectionClick={onCollectionClick}
          />
        </div>
      );
    }
  }
}

const mapStateToProps = (state: StoreState) => {
  const {
    authPrompt,
    accountName,
    loggedIn,
    dashboard,
    openAlert,
    alertMessage
  } = state.authentication;

  return {
    authPrompt,
    accountName,
    loggedIn,
    dashboard,
    openAlert,
    alertMessage
  };
};

const mapDispatchToProps = (
  dispatch: ThunkDispatch<StoreState, void, AuthAction | RouterAction>
) => {
  return {
    onAlert: () => dispatch(closeAlert()),
    onLogInPrompt: () => dispatch(logInPrompt()),
    onLogOut: () => dispatch(logOut()),
    onClosePrompt: () => dispatch(closePrompt()),
    onAuthAccount: (name: string, pass: string) => dispatch(logIn(name, pass)),
    onCollectionClick: (id: string) => dispatch(push(`/collections/${id}`)),
    onFileClick: (id: string) => dispatch(push(`/files/${id}`)),
    onHomeClick: () => dispatch(push("/"))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DashboardContainer);
