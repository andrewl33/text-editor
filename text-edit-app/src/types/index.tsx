export interface StoreState {
  authentication: AuthStoreState;
  collection: CollectionStoreState;
  editor: EditorStoreState;
  router: RouterStoreState;
}

/**
 *
 * Connected Components
 *
 */

// authentication
export interface AuthStoreState {
  loggedIn: boolean;
  accountName?: string;
  dashboard?: DashboardProps;
  openAlert: boolean;
  alertMessage?: string;
  authPrompt: boolean;
}

// Collection
export interface CollectionStoreState extends CollectionComponentProps {
  openAlert: boolean;
  alertMessage: string;
  collectionPrompt: boolean;
}

export interface CollectionComponentProps {
  items: Item[] | null;
  name: string;
  createDate: string;
  users: string[];
  isLocked: boolean;
}

// Editor
export interface EditorStoreState {
  codeText: string;
  tags: string[];
  isLoading: boolean;
  isNewPage: boolean;
  hasAuth: boolean;
  isLocked: boolean;
  isSaved: boolean;
  openAlert: boolean;
  alertMessage: string;
  filePrompt: boolean;
  name: string;
  createDate: string;
  users: string[];
}

export interface EditorProps extends EditorStoreState {
  onBatchUpdate: (codeText: string) => any;
  onCodeChange: () => any;
  onAlert: () => any;
  onLock: () => any;
  onMount: () => any;
  onNew: () => any;
  onShare: () => any;
  onAuthFile?: () => any;
  onAuthAccount?: () => any;
  onRemoveTag?: (tagName: string) => any;
  onAddTag?: (tagName: string) => any;
  onNameChange: (name: string) => any;
  pathname: string;
  authPrompt: boolean;
}

// home
export interface HomeProps {
  onNewFile: () => {};
  onNewCollection: () => {};
  onLogin: (name: string, pass: string) => {};
  toDashboard: () => {};
  loggedIn: boolean;
}

// user
export interface UserStoreState extends UserProps {
  dashboard: DashboardProps;
}

export interface UserProps {
  loggedIn: boolean;
  accountName?: string;
}

export interface DashboardProps {
  collections: Item[] | null;
  files: Item[] | null;
}

// router
export interface RouterStoreState {
  action: string;
  location: {
    hash: string;
    pathname: string;
    search: string;
  };
}

/**
 *
 * Helpers
 *
 */

// header
export interface HeaderProps extends PromptComponentProps {
  onNew?: () => {};
  onLock?: () => {};
  onShare?: () => {};
  onAlert: () => {};
  accountName?: string;
  loggedIn: boolean;
  pathname?: string;
  isShareable?: boolean;
  openAlert: boolean;
  alertMessage?: string;
  pageName: string;
  onPrompt?: boolean;
}

// header/prompts
export interface PromptComponentProps {
  getAccountCredentials?: (accountName: string, password: string) => {};
  getPassword?: (password: string) => {};
  prompt?: "Login" | "Create Account" | "Private Collection" | "Private File";
}

export interface Item {
  uuid: string;
  name: string;
  tags?: string[];
  date: string;
}

// lists
export interface ListProps {
  header: "Collections" | "Files";
  items: Item[] | null;
}

// sidebar
export interface SidebarProps {
  name: string;
  pageType: "file" | "collection";
  tagList?: string[];
  allTagsList?: string[];
  createDate: string;
  isPrivate: boolean;
  users: string[];
  onRemoveTag?: (tagName: string) => any;
  onAddTag?: (tagName: string) => any;
  onNameChange: (name: string) => any;
}
