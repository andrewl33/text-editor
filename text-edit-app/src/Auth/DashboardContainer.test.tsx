import * as React from "react";
import * as renderer from "react-test-renderer";
import { PromptType } from "../types";
import { DashboardContainer } from "./DashboardContainer";
const authProps = {
  authPrompt: false,
  accountName: "",
  loggedIn: false,
  dashboard: { collections: [], files: [] },
  openAlert: false,
  alertMessage: "default"
};

const dispatchProps = {
  onAlert: jest.fn(),
  onLogInPrompt: jest.fn(),
  onLogOut: jest.fn(),
  onClosePrompt: jest.fn(),
  onAuthAccount: jest.fn(),
  onCollectionClick: jest.fn(),
  onFileClick: jest.fn(),
  onHomeClick: jest.fn(),
  onMount: jest.fn(),
  toGradingPage: jest.fn()
};

const staticProps = {
  isShareable: false,
  pageName: "Dashboard",
  prompt: "Login" as PromptType
};

const props = {
  ...authProps,
  ...dispatchProps,
  ...staticProps
};

describe("DashboardContainer", () => {
  test("not logged in renders correctly", () => {
    const component = renderer.create(<DashboardContainer {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("logged in renders correctly", () => {
    const loggedInProps = { ...props, loggedIn: true, accountName: "test" };

    const component = renderer.create(
      <DashboardContainer {...loggedInProps} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
