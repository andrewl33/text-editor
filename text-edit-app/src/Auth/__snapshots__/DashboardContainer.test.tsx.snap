// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardContainer logged in renders correctly 1`] = `
<div>
  <header>
    <div
      className="ui huge inverted attached menu"
      primary="true"
    >
      <a
        className="item"
        onClick={[Function]}
      >
        Text Edit
      </a>
      <div
        className="item"
        onClick={[Function]}
      >
        Dashboard
      </div>
      <div
        className="right menu"
      >
        <div
          className="item"
          onClick={[Function]}
        >
          <div>
            <div
              aria-expanded={false}
              className="ui button floating labeled dropdown icon"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="listbox"
              style={
                Object {
                  "backgroundColor": "#2185d0",
                  "color": "#fff",
                  "fontSize": 17,
                }
              }
              tabIndex={0}
            >
              <div
                aria-live="polite"
                className="text"
                role="alert"
              >
                test
              </div>
              <i
                aria-hidden="true"
                className="user icon"
                onClick={[Function]}
              />
              <div
                className="menu transition"
              >
                <div
                  className="item"
                  onClick={[Function]}
                  role="option"
                >
                  Dashboard
                </div>
                <div
                  className="item"
                  onClick={[Function]}
                  role="option"
                >
                  Log out
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
  <div>
    <div>
      <table
        className="ui blue large celled inverted middle aligned table"
        divided="true"
        relaxed="true"
        style={
          Object {
            "borderRadius": 0,
            "color": "rgba(255,255,255,.9)",
          }
        }
      >
        <thead
          className=""
        >
          <tr
            className=""
          >
            <th
              className=""
              colSpan="2"
            >
              Collections
            </th>
            <th
              className=""
              colSpan="2"
            >
              Dates
            </th>
          </tr>
        </thead>
        <tbody
          className=""
        />
      </table>
    </div>
    <div>
      <table
        className="ui blue large celled inverted middle aligned table"
        divided="true"
        relaxed="true"
        style={
          Object {
            "borderRadius": 0,
            "color": "rgba(255,255,255,.9)",
          }
        }
      >
        <thead
          className=""
        >
          <tr
            className=""
          >
            <th
              className=""
              colSpan="2"
            >
              Files
            </th>
            <th
              className=""
            >
              Tags
            </th>
            <th
              className=""
              colSpan="2"
            >
              Dates
            </th>
          </tr>
        </thead>
        <tbody
          className=""
        />
      </table>
    </div>
  </div>
</div>
`;

exports[`DashboardContainer not logged in renders correctly 1`] = `
<div>
  <header>
    <div
      className="ui huge inverted attached menu"
      primary="true"
    >
      <a
        className="item"
        onClick={[Function]}
      >
        Text Edit
      </a>
      <div
        className="item"
        onClick={[Function]}
      >
        Dashboard
      </div>
      <div
        className="right menu"
      >
        <div
          className="item"
          onClick={[Function]}
        >
          <div>
            <div
              aria-expanded={false}
              className="ui button floating labeled dropdown icon"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="listbox"
              style={
                Object {
                  "backgroundColor": "#2185d0",
                  "color": "#fff",
                  "fontSize": 17,
                }
              }
              tabIndex={0}
            >
              <div
                aria-live="polite"
                className="text"
                role="alert"
              >
                Account
              </div>
              <i
                aria-hidden="true"
                className="user icon"
                onClick={[Function]}
              />
              <div
                className="menu transition"
              >
                <div
                  className="item"
                  onClick={[Function]}
                  role="option"
                >
                  Log in
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
  <h1
    style={
      Object {
        "color": "rgba(255,255,255,.9)",
      }
    }
  >
    Not logged in!
  </h1>
</div>
`;
